polar-core/src/partial/partial.rs

```diff
@@ -588,8 +601,8 @@ mod test {
 
         let mut q = p.new_query_from_term(term!(call!("h", [sym!("x"), sym!("y")])), false);
         assert_partial_expressions!(next_binding(&mut q)?,
-            "x" => "y.x matches X{}",
-            "y" => "_this.x matches X{}"
+            "x" => "_this matches X{} and y.x = _this",
+            "y" => "x matches X{} and _this.x = x"
```

Comment:
little worried about changes like this trickling into the django-oso / sqlalchemy-oso constraint transformation code — do we have tests in those libraries for this behavior?

Reply:
No, what should they test?

Reply:
At a minimum I'd feel more confident if we had analogous test cases for `h(x2: X, _: {x: x2});` and `foo(u: User, x: Post) if x.user = u;`, which seem to be the two cases in the core that were impacted by this change. What do you think?

Reply:
But what are they actually testing? Are they identical to the core tests?

Reply:
Ok, I've looked into this. SQLAlchemy and Django can't handle the partials that come out of rules like this, because they have multiple unbound variables. Since the adapters are intended to handle queries over one free variable, I don't think this is likely to cause any issues. But I added tests to each library to check that a partial with both an isa and a dot operation works & produces the expected result. What do you think?

Reply:
![exceeds expectations](https://smallbizclub.com/wp-content/uploads/2019/09/Are-You-Meeting-and-Exceeding-Customer-Expectations.jpg)

Reply:
ty gj!!